tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -506,6 +494,91 @@ shared_ptr<CKKSTensor> CKKSTensor::polyval_inplace(
     return shared_from_this();
 }
 
+shared_ptr<CKKSTensor> CKKSTensor::dot_inplace(
+    const shared_ptr<CKKSTensor>& other) {
+    auto this_shape = this->shape();
+    auto other_shape = other->shape();
+
+    if (this_shape.size() == 1) {
+        if (other_shape.size() == 1) {  // 1D-1D
+            // inner product
+            this->mul_inplace(other);
+            this->sum_inplace();
+            return shared_from_this();
+        } else if (other_shape.size() == 2) {  // 1D-2D
```

Comment:
For broadcasting there are more checks needed: the dimensions must be equal or one of them = 1.

Maybe leave the condition 
```
if (this_shape.size() == 1 && other_shape.size() == 1)
```
and a TODO, if the broadcasting is not intended for this PR. Or throw an exception in this case too.

Reply:
it's actually just a placeholder for now and I would only need to send an error saying that this shapes aren't supported yet

Reply:
still, it should throw an exception.